oak/src/oak/server_test.cpp

```diff
@@ -0,0 +1,214 @@
+
+/**
+ * @brief Tests that stopping a not-listening server results in an exception
+ *
+ */
+TEST(oak_server, fail_not_listening) {
+  try {
+    oak::Server server;
+    server.stop();
+
+    FAIL() << "Expected std::runtime_error";
+  } catch (std::runtime_error) {
+    SUCCEED();
+  } catch (std::exception) {
+    FAIL() << "Expected std::runtime_error";
+  }
+}
+
+/**
+ * @brief Tries to GET a non-existing endpoint
+ */
+TEST(oak_server, get_nohandler) {
+  // Create server
+  auto server = create_server();
+  // Read a non-existing endpoint
+  auto address_ = server->address();
+  auto port_ = server->port();
+  const auto cmd =
+      boost::str(boost::format("curl -X GET %1%:%2%/test 2>/dev/null") % address_ % port_);
+  auto result = exec(cmd);
+
+  // Compare result
+  EXPECT_STREQ(result.c_str(), "{\"endpoints\":[],\"error\":\"cannot find handler\"}");
+}
+
+/**
+ * @brief Tries to GET an endpoint
+ */
+TEST(oak_server, get_handler) {
+  // Create server
+  auto server = create_server();
+  // Create registrar
+  oak::StaticRegistrar registrar(server.get(), "", nullptr);
+  // Register one pseudo-endpoint
+  std::vector<std::string> data{
+      "1",
+      "2",
+      "3",
+  };
+  registrar.register_handler("/simulators", cloe::handler::StaticJson(data));
+
+  // Read the pseudo-endpoint
+  auto address_ = server->address();
+  auto port_ = server->port();
+  const auto cmd =
+      boost::str(boost::format("curl -X GET %1%:%2%/simulators 2>/dev/null") % address_ % port_);
+  auto result = exec(cmd);
+
+  // Compare result
+  EXPECT_STREQ(result.c_str(), "[\"1\",\"2\",\"3\"]");
+}
+
+/**
+ * @brief Tries to POST to an endpoint which echos the data back
+ */
+TEST(oak_server, post_handler) {
+  // Create server
+  auto server = create_server();
+  // Create registrar
+  oak::StaticRegistrar registrar(server.get(), "", nullptr);
+  // Register one pseudo-endpoint
+  std::vector<std::string> data{
+      "1",
+      "2",
+      "3",
+  };
+  registrar.register_handler(
+      "/echo", [this](const cloe::Request& request, cloe::Response& response) {
+        switch (request.method()) {
+          case cloe::RequestMethod::POST: {
+            try {
+              const auto json = request.as_json();
+              response.write(json);
+              response.set_status(cloe::StatusCode::OK);
+            } catch (const std::exception& ex) {
+              response.bad_request(cloe::Json{
+                  {"error", ex.what()},
+              });
+            }
+            break;
+          }
+          default: {
+            response.not_allowed(cloe::RequestMethod::POST,
+                                 cloe::Json{
+                                     {"error", "only GET or POST method allowed"},
+                                 });
+          }
+        }
+      });
+
+  // Read the pseudo-endpoint
+  auto address_ = server->address();
+  auto port_ = server->port();
+  const auto cmd =
```

Comment:
Is there a reason we can't just use the `fmt` library we are using in the rest of the project for all string formatting?